controllers/metric_controller.go

```diff
@@ -191,6 +192,16 @@ func (r *MetricReconciler) collectMetrics(ctx context.Context, t *redskyv1beta1.
 		return controller.RequeueConflict(err)
 	}
 
+	// Wait until all metrics have been collected to fail the trial for an out of bounds metric
+	for i := range t.Spec.Values {
+		v := &t.Spec.Values[i]
+		if err := validation.CheckMetricBounds(metrics[v.Name], v); err != nil {
+			trial.ApplyCondition(&t.Status, redskyv1beta1.TrialFailed, corev1.ConditionTrue, "MetricBound", err.Error(), probeTime)
```

Comment:
does failing a trial here prevent us from posting the metrics?  
I'm assuming the metrics would still get sent since the metrics have already been evaluated and exist on the trial

Reply:
It does prevent us from posting the actual metric values to the API (we only send the actual values if the trial is not failed). I don't know if the backend would be OK with us sending failed trials that include values (but maybe that should also be part of non-optimized metrics?). @jdplatt ?